pldp/survey/models.py

```diff
@@ -0,0 +1,149 @@
+                                                  "the specific time of the "
+                                                  "survey count string that "
+                                                  "may have impacted the "
+                                                  "results."))
+    representation = models.CharField(max_length=8,
+                                      choices=[('absolute', _("Absolute")),
+                                               ('relative', _("Relative"))],
+                                      help_text=_("Indicate whether the data "
+                                                  "collected a total of the "
+                                                  "people present within the "
+                                                  "survey count time or a "
+                                                  "representative sample"),
+                                      null=True,
+                                      blank=True)
+    microclimate = models.CharField(max_length=255,
+                                    null=True,
+                                    blank=True,
+                                    help_text=_("Perceived weather condition "
+                                                "on the specific survey "
+                                                "location."))
+    temperature_c = models.IntegerField(null=True,
+                                        blank=True,
+                                        help_text=_("Official temperature "
+                                                    "measured in the survey "
+                                                    "location at the time of "
+                                                    "the survey count."))
+    method = models.CharField(max_length=20,
+                              choices=METHOD_CHOICES,
+                              help_text=_("Description of the survey count "
+                                          "method"))
+
+    def totals(self, query_filters=[]):
+
+        # Takes in a list of ways you'd like to filter the survey results and
+        # returns a count by component, by choice
+
+        pass
+
+
+class SurveyRow(models.Model):
+    survey = models.ForeignKey(Survey, on_delete=models.CASCADE)
+    total = models.IntegerField(help_text=_("Indicate the number of people "
+                                            "counted within the row. This "
+                                            "field is not an ID, but it should "
+                                            "be included with every survey."),
+                                default=1)
+
+    @property
+    def response(self):
+        response = {}
+        for component in self.surveycomponent_set.all():
+            response[component.type_of_component] = component.choice
+
+        return response
+
+
+class AbstractSurveyComponent(models.Model):
```

Comment:
Last I talked to Eric, he wasn't sure about this structure. It makes sense to me to have an abstracted survey component class that the actual components are based on, but I'm not sure why gender and age are included here. The PLDP doesn't include a huge number of possible components (`survey_gender, survey_age, survey_mode, survey_groups, survey_posture, survey_activities, survey_objects, survey_geotag`) and I'm inclined to say that we should create them as their own classes.

Reply:
You can take the lead on this one!